core/scripts/server.js

```diff
@@ -120,10 +121,18 @@ app.use('/service-worker.js', serve('dist/service-worker.js', false, {
     res.set('Content-Type', 'text/javascript; charset=UTF-8')
   }
 }))
-
+/** @deprecated in 2.0  in favour of module-based mechanism */
 const serverExtensions = require(resolve('src/server'))
 serverExtensions.registerUserServerRoutes(app)
 
+config.serverSideModules.forEach(serverModule => {
+  if (Array.isArray(serverModule)) {
+    require(resolve(serverModule[0] + '/server.js'))(app, serverModule[1])
+  } else {
```

Comment:
Cool, but few questions:
- Does it support ES6/TS in server modules? Otherwise no possibility to reuse VS code and libs
- do we really need this config variable? It’s one time operation so we can probably traverse core/modules and src/modules(?)
- we need an example module in the module template

Reply:
@pkarw 

I can say for the first point that it does not support typescript, but that would be a really nice feature.

Reply:
yeah, would be great to have these modules in separate dir or register in some index.js file like modules, this index.js would always be consumed on the server side. I'm afraid that config option could be overcomplicated.

And example +1 :)

Reply:
@pkarw @patzick  imho we need to if you want to include modules from node_modules as well. We can't really assume that their always named `vsf-something`. 
Separate index.js is an optiion as well but for sure we can't just traverse the file tree. Please pick either config or file.

About es6 tbh I don't know, will check it. It works exactly the same as server.js extensions

Reply:
so I'm voting for registering in a file instead of config :)